From 9e2309837bd3184b9a3157adabd6b5b17aa2b990 Mon Sep 17 00:00:00 2001
From: Lin Ruizhe <linruizhe@huawei.com>
Date: Fri, 31 Dec 2021 15:59:23 +0800
Subject: [PATCH] irq: Add kabi_reserve in irq
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 9e2309837bd3184b9a3157adabd6b5b17aa2b990


hulk inclusion
bugzilla: https://gitee.com/openeuler/kernel/issues/I4MZU1
CVE: NA

---------------------------

Add kabi_reserve in irq_common_data irq_chip irq_chip_type
in irq.h

Signed-off-by: Lin Ruizhe <linruizhe@huawei.com>
Reviewed-by: Liao Chang <liaochang1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/irq.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/linux/irq.h b/include/linux/irq.h
index 607bee9271bd..34a65cb1cd9e 100644
--- a/include/linux/irq.h
+++ b/include/linux/irq.h
@@ -19,6 +19,7 @@
 #include <linux/topology.h>
 #include <linux/io.h>
 #include <linux/slab.h>
+#include <linux/kabi.h>
 
 #include <asm/irq.h>
 #include <asm/ptrace.h>
@@ -156,6 +157,9 @@ struct irq_common_data {
 #ifdef CONFIG_GENERIC_IRQ_IPI
 	unsigned int		ipi_offset;
 #endif
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+
 };
 
 /**
@@ -550,6 +554,7 @@ struct irq_chip {
 	void		(*irq_nmi_teardown)(struct irq_data *data);
 
 	unsigned long	flags;
+	KABI_RESERVE(1)
 };
 
 /*
@@ -1015,6 +1020,7 @@ struct irq_chip_type {
 	u32			type;
 	u32			mask_cache_priv;
 	u32			*mask_cache;
+	KABI_RESERVE(1)
 };
 
 /**
-- 
2.26.2

